qhana_plugin_registry/db/models/templates.py

```diff
@@ -231,6 +267,34 @@ class TemplateTab(IdMixin, ExistsMixin, NameDescriptionMixin):
     def plugin_filter(self) -> dict:
         return json.loads(self.filter_string)
 
+    @classmethod
+    def get_by_name(cls, name: str) -> "Optional[TemplateTab]":
+        q = select(cls).filter(cls.name == name)
+        found_tab: Optional[TemplateTab] = DB.session.execute(q).scalar_one_or_none()
+        return found_tab
+
+    @classmethod
+    def get_or_create(cls, tab: dict) -> "TemplateTab":
+        found_tab: Optional[TemplateTab] = cls.get_by_name(tab["name"])
+        if found_tab is None:
+            found_tab = cls(
+                name=tab["name"],
+                description=tab["description"],
+                sort_key=tab["sort_key"],
+                location=tab["location"],
+                filter_string=json.dumps(tab["filter"]),
+                template=tab["template"],
+            )
+            DB.session.add(found_tab)
+            DB.session.commit()
```

Comment:
you can have an autocommit in such methods, but please use a default function param to controll if the commit actually happens:

```python
def do_db_stuff(data, commit: bool=True):
    stuff(data)
    
    if commit:
        DB.session.commit()
```

Maybe even use `False` as the default value for explicit opt in to such autocommits. (This is how it is handled in the plugin runner)